packages/lab/src/__tests__/__e2e__/list-next/ListNext.cy.tsx

```diff
@@ -0,0 +1,346 @@
+            cy.findByRole("option", { name: ITEMS[1].label }).should(
+              "have.class",
+              "saltListItemNext-focused"
+            );
+            cy.findByRole("listbox").blur(); // remove focus
+            cy.findByRole("listbox").focus(); // focus again
+            // focus should be back on the previously focused item
+            cy.findByRole("option", { name: ITEMS[1].label }).should(
+              "have.class",
+              "saltListItemNext-focused"
+            );
+          });
+        });
+
+        describe("WHEN an item was previously selected, unless focus has moved", () => {
+          it("THEN re-focus on previously selected item", () => {
+            cy.mount(<SingleSelectList />);
+
+            cy.findByRole("listbox").focus();
+            cy.realPress("ArrowDown");
+            cy.realPress("ArrowDown");
+            cy.realPress("Enter");
+            // list item selected
+            cy.findByRole("option", { name: ITEMS[2].label }).should(
+              "have.attr",
+              "aria-selected",
+              "true"
+            );
+            cy.findByRole("listbox").blur(); // remove focus
+            cy.findByRole("listbox").focus(); // focus again
+            // focus should be back on the previously selected item
+            cy.findByRole("option", { name: ITEMS[2].label }).should(
+              "have.class",
+              "saltListItemNext-focused"
+            );
+          });
+        });
+
+        describe("WHEN focus is on the last item", () => {
+          it("THEN focus does not wrap on arrow down", () => {
+            cy.mount(<SingleSelectList />);
+
+            cy.findByRole("listbox").focus();
+            cy.realPress("ArrowDown");
+            cy.realPress("ArrowDown");
+            cy.realPress("ArrowDown");
+            // focus on last list item
+            cy.findByRole("option", { name: ITEMS[2].label }).should(
+              "have.class",
+              "saltListItemNext-focused"
+            );
+            cy.realPress("ArrowDown");
+            // focus do not wrap
+            cy.findByRole("option", { name: ITEMS[2].label }).should(
+              "have.class",
+              "saltListItemNext-focused"
+            );
+          });
+        });
+
+        describe("WHEN focus is on the first item", () => {
+          it("THEN focus does not wrap on arrow up", () => {
+            cy.mount(<SingleSelectList />);
+
+            cy.findByRole("listbox").focus();
+            cy.realPress("ArrowUp");
+            // focus do not wrap
+            cy.findByRole("option", { name: ITEMS[0].label }).should(
+              "have.class",
+              "saltListItemNext-focused"
+            );
+          });
+        });
+
+        describe("WHEN on Home key press", () => {
+          it("THEN focus on first list item", () => {
+            cy.mount(<SingleSelectList />);
+
+            cy.findByRole("listbox").focus();
```

Comment:
maybe select another item first and then press Home. This test might still pass if there is no handler on Home. Focusing the list will focus the first item anyway, right?

Reply:
yeah that makes sense - focusing the list will focus first item yes. Updated the test now